### PR TITLE
Setup docker-compose

### PR DESCRIPTION
**Pull request by bmarkons**
Once upon a time there was an idea of docker-compose https://github.com/ruby-bench/ruby-bench-docker/issues/13 :smile:

I was trying to debug issue with discourse query benchmark. We encountered unexpectedly similar performance of ActiveRecord and Sequel flavors for that benchmark. Therefore I assumed something is bad in docker setup and decided to try with docker-compose.

Results I got locally with this setup : 

- Raw : 230 ips
- Sequel : 190 ips
- ActiveRecord : 150

Overhead with ActiveRecord is 2x greater than with Sequel.

### CODE REVIEW

File: rails/rails_trunk/rails_benchmarks/runner [hunk 1]
```diff
@@ -1,20 +1,10 @@
...
-done
-
-dropdb --if-exists -h "$POSTGRES_PORT_5432_TCP_ADDR" -p "$POSTGRES_PORT_5432_TCP_PORT" -U postgres rubybench
-createdb -h "$POSTGRES_PORT_5432_TCP_ADDR" -p "$POSTGRES_PORT_5432_TCP_PORT" -U postgres rubybench
-echo "DROP DATABASE IF EXISTS rubybench" | mysql -u root -h "$MYSQL_PORT_3306_TCP_ADDR" -P "MYSQL_PORT_3306_TCP_PORT"
-echo "CREATE DATABASE rubybench" | mysql -u root -h "$MYSQL_PORT_3306_TCP_ADDR" -P "MYSQL_PORT_3306_TCP_PORT"
+dropdb --if-exists -h postgres -U postgres rubybench
+createdb -h postgres -U postgres rubybench
+echo "DROP DATABASE IF EXISTS rubybench" | mysql -u root -h mysql
+echo "CREATE DATABASE rubybench" | mysql -u root -h mysql
```
Comment (bmarkons) [inline rails/rails_trunk/rails_benchmarks/runner:16]: I am not sure though if recreating database should occur before (or after) every benchmark? @noahgibbs

Comment (bmarkons) [reply rails/rails_trunk/rails_benchmarks/runner:16]: We are recreating only tables used in certain benchmark but not entire database. I have feeling that we should drop and create whole database before each bench. I am not sure if having two many tables in database affects performance even though you are not using them all?

Comment (noahgibbs) [reply rails/rails_trunk/rails_benchmarks/runner:16]: Having tables you're not using doesn't significantly affect performance during a benchmark unless you're doing something unusual (like enumerating all the tables.) ActiveRecord classes you aren't using will add a bit of overhead, but that's true Ruby-side as well.

Recreating the database would primarily be a way of making sure tables don't have extra records in them we're not using, which would make more difference in some cases.

Comment (bmarkons) [reply rails/rails_trunk/rails_benchmarks/runner:16]: We are making sure we don't have extra records in tables before every run by recreating tables, though I think we could recreate entire database before each benchmark.

Comment (noahgibbs) [reply rails/rails_trunk/rails_benchmarks/runner:16]: Sure. I think you're doing fine on that. I just mean that extra records in tables you use are slow (that's fine, you're taking care of that.) But extra tables you don't use are basically free.

File: pg/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+version: '3'
+services:
+  postgres:
+    image: postgres:9.6
```
Comment (tgxworld) [inline pg/docker-compose.yml:4]: Did this affect the performance of our benchmark? We'll probably have to invalidate all benchmarks that were ran with Postgres 9.5

File: pg/runner [hunk 3]
```diff
@@ -1,13 +1,12 @@
 #!/bin/bash
 set -e
 
-while ! exec 6<>/dev/tcp/${POSTGRES_PORT_5432_TCP_ADDR}/${POSTGRES_PORT_5432_TCP_PORT}; do
-  echo "$(date) - still trying to connect to Postgres server"
+until pg_isready -h postgres; do
```
Comment (tgxworld) [inline pg/runner:4]: :+1:

**Comment by tgxworld** [comment]
:clap: loving the changes here

File: pg/docker-compose.yml (hunk 2 above)
Comment (bmarkons) [reply pg/docker-compose.yml:4]: Probably invalidation is the right thing to do.

**Pull request merged**